packages/yarnpkg-pnpify/sources/Hoister.ts

```diff
@@ -0,0 +1,289 @@
+      addedIds.add(pkgId);
+      const deps = new Set<PackageId>();
+      const peerDeps = new Set<PackageId>();
+      packageTree[pkgId] = {deps, peerDeps};
+
+      for (const [name, reference] of pkg.packageDependencies) {
+        if (reference) {
+          const locator = typeof reference === 'string' ? {name, reference} : {name: reference[0], reference: reference[1]};
+          const depPkgId = getPackageId(locator);
+          if (depPkgId) {
+            if (pkg.packagePeers) {
+              if (pkg.packagePeers.has(name)) {
+                peerDeps.add(depPkgId);
+              } else {
+                deps.add(depPkgId);
+              }
+            } else {
+              if (parentDepIds.has(depPkgId) && depPkgId !== pkgId) {
+                peerDeps.add(depPkgId);
+              } else {
+                deps.add(depPkgId);
+              }
+            }
+          }
+        }
+      }
+
+      const allDepIds = new Set(Array.from(deps).concat(Array.from(peerDeps)));
+      for (const depId of allDepIds) {
+        const depPkg = packageInfos[depId];
+        if (depPkg && !addedIds.has(depId)) {
+          addPackageToTree(depPkg, depId, allDepIds);
+        }
+      }
+    };
+
+    const assignedIds = new Set<PackageId>();
+    const assignPackageIds = (pkgId: PackageId) => {
+      assignedIds.add(pkgId);
+      const pkg = packageInfos[pkgId];
+      for (const [name, reference] of pkg.packageDependencies) {
+        if (reference) {
+          const locator = typeof reference === 'string' ? {name, reference} : {name: reference[0], reference: reference[1]};
+          const depPkg = pnp.getPackageInformation(locator);
+          if (depPkg) {
+            const depPkgId = assignPackageId(locator, depPkg);
+            if (!assignedIds.has(depPkgId)) {
+              assignPackageIds(depPkgId);
+            }
+          }
+        }
+      }
+    };
+
+    const pkg = pnp.getPackageInformation(pnp.topLevel)!;
+    const topLocatorKey = getLocatorKey(pnp.topLevel);
+    for (const locator of pnpRoots) {
+      if (getLocatorKey(locator) !== topLocatorKey) {
+        pkg.packageDependencies.set(locator.name!, locator.reference);
+      }
+    }
+    const pkgId = assignPackageId(pnp.topLevel, pkg);
+    assignPackageIds(pkgId);
+
+    addPackageToTree(pkg, pkgId, new Set<number>());
+
+    return {packageTree, packages, locators};
+  }
+
+  /**
+   * Converts hoisted tree to node modules map
+   *
+   * @param pnp PnP API
+   * @param hoistedTree hoisted package tree from `RawHoister`
+   * @param locators locators
+   *
+   * @returns node modules map
+   */
+  private buildNodeModulesTree(pnp: PnpWalkApi, hoistedTree: HoistedTree, locators: PackageLocator[]): NodeModulesTree {
+    const tree: NodeModulesTree = new Map();
+
+    // Hard link Regex heuristics, needed for old PnP API versions support
+    const HARD_LINK_REGEX = /((\.zip|-integrity)[\\/]node_modules[\\/]|-virtual-[0-9a-f]{10})/;
```

Comment:
I'm still not sure we should support old ones tbf ... people already using PnP don't have those issues otherwise they wouldn't be using it, and people would want to adopt PnP will likely use the v2 anyway 🤔

Reply:
Done, removed